packages/typegpu/src/prepareDispatch.ts

```diff
@@ -0,0 +1,84 @@
+import { builtin } from './builtin.ts';
+import { computeFn } from './core/function/tgpuComputeFn.ts';
+import { fn } from './core/function/tgpuFn.ts';
+import type { TgpuRoot } from './core/root/rootTypes.ts';
+import { u32 } from './data/numeric.ts';
+import { vec3f, vec3u } from './data/vector.ts';
+import type { v3u } from './data/wgslTypes.ts';
+import { any, ge } from './std/boolean.ts';
+import { ceil } from './std/numeric.ts';
+
+/**
+ * Changes the given array to a vec of 3 numbers, filling missing values with 1.
+ */
+function sanitizeArray(arr: readonly number[]): v3u {
```

Comment:
I don't think this name is very fitting since there is no much sanitization going on. Maybe something like:
```suggestion
function toVec3(arr: readonly number[]): v3u {
```
Also I just noticed we are not safeguarding against someone passing in `0`.